File: easypermissions/src/main/java/za/co/bwmuller/easypermissions/AppCompatCallbackActivity.java
package za.co.bwmuller.easypermissions;

import android.support.v7.app.AppCompatActivity;

import java.util.ArrayList;

/**
 * Created by Bernhard Müller on 8/23/2017.
 */

public class AppCompatCallbackActivity extends AppCompatActivity {
    ArrayList<ActivityListener> listenerList = new ArrayList<>();

    public void addListener(ActivityListener listener) {
        listenerList.add(listener);
    }

    public boolean removeListener(ActivityListener listener) {
        return listenerList.remove(listener);
    }
}

File: easypermissions/src/main/java/za/co/bwmuller/easypermissions/AppCompatCallbackActivity.kt
package za.co.bwmuller.easypermissions

import androidx.appcompat.app.AppCompatActivity
import java.util.ArrayList

/**
 * Created by Bernhard Müller on 8/23/2017.
 */
open class AppCompatCallbackActivity : AppCompatActivity() {

    var listenerList = ArrayList<ActivityListener>()
    fun addListener(listener: ActivityListener) {
        listenerList.add(listener)
    }

    fun removeListener(listener: ActivityListener): Boolean {
        return listenerList.remove(listener)
    }
}